src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Reliability/SqlConfigurableRetryLogicManager.CreateProvider.cs

```diff
@@ -0,0 +1,334 @@
+                    retryOption.TransientErrors = _configSection.TransientErrors.Split(',').Select(x => Convert.ToInt32(x)).ToList();
+                }
+
+                // Prepare the authorized SQL statement just for SqlCommands
+                if (_configSection is SqlConfigurableRetryCommandConfigurationSection cmdSection && !string.IsNullOrEmpty(cmdSection.AuthorizedSqlCondition))
+                {
+                    retryOption.AuthorizedSqlCondition = (x) => Regex.IsMatch(x, cmdSection.AuthorizedSqlCondition);
+                }
+                SqlClientEventSource.Log.TryTraceEvent("<sc.{0}.{1}|INFO> Successfully created a {2} object to use on creating a retry logic provider from the section '{3}'.",
+                                                       s_typeName, methodName, nameof(SqlRetryLogicOption), sectionName);
+
+                // Extract the SqlRetryLogicBaseProvider object by the given information
+                var provider = ResolveRetryLogicProvider(_configSection.RetryLogicType, _configSection.RetryMethod, retryOption);
+                SqlClientEventSource.Log.TryTraceEvent("<sc.{0}.{1}|INFO> Successfully created a {2} object from the section '{3}'.",
+                                                       s_typeName, methodName, nameof(SqlRetryLogicBaseProvider), sectionName);
+                return provider;
+            }
+            catch (ConfigurationErrorsException e)
+            {
+                // Don't throw an error for invalid config files
+                SqlClientEventSource.Log.TryTraceEvent("<sc.{0}.{1}|INFO>: Unable to load custom {2}. ConfigurationManager failed to load due to the configuration errors: {3}",
+                                                       s_typeName, methodName, sectionName, e);
+            }
+            catch (Exception e)
+            {
+                SqlClientEventSource.Log.TryTraceEvent("<sc.{0}.{1}|INFO> {2}",
+                                                       s_typeName, methodName, e);
+            }
+            SqlClientEventSource.Log.TryTraceEvent("<sc.{0}.{1}|INFO> Due to the exception occurrence, the default none-retriable logic will apply",
+                                                    s_typeName, methodName);
+            // Return default provider if an exception occured.
+            return SqlConfigurableRetryFactory.CreateNoneRetryProvider();
+        }
+
+        private static SqlRetryLogicBaseProvider ResolveRetryLogicProvider(string configurableRetryType, string retryMethod, SqlRetryLogicOption option)
+        {
+            var methodName = MethodBase.GetCurrentMethod().Name;
+            SqlClientEventSource.Log.TryTraceEvent("<sc.{0}.{1}|INFO> Entry point.", s_typeName, methodName);
+
+            if (string.IsNullOrEmpty(retryMethod))
+            {
+                throw new ArgumentNullException($"Failed to create {nameof(SqlRetryLogicBaseProvider)} object because the {nameof(retryMethod)} value is null or empty.");
```

Comment:
All exception messages need to come from Resource Builder (Strings.resx) and preferably built via SqlUtil.

Reply:
All of the exceptions in this class handles internally to log with Event Source. I've added a note over the class definition to explain it. I'll add them to the Resource Builder If you believe it's required.